zproject/settings.py

```diff
@@ -52,6 +52,30 @@ def get_config(section: str, key: str, default_value: Optional[Any]=None) -> Opt
         return config_file.get(section, key)
     return default_value
 
+def get_from_file_if_exists(path: str) -> str:
+    if os.path.exists(path):
+        with open(path, "r") as f:
+            return f.read()
+    else:
+        return ''
+
+def parse_cryptographic_key_to_single_line(input_key: str) -> str:
+
+    def is_line_redundant(line: str) -> bool:
+        # Check for lines like "-----BEGIN PRIVATE KEY-----" and similar.
+        line = line.strip()
+        if line.startswith('--') and line.endswith('--'):
+            return True
+        return False
+
+    lines = input_key.strip().split('\n')
+    if is_line_redundant(lines[0]):
+        lines = lines[1:]
+    if is_line_redundant(lines[-1]):
+        lines = lines[:-1]
+
+    return ''.join(lines)
```

Comment:
If we don't need this, it'd be nice to not have it, as it's a decent amount of complexity.

Reply:
I got rid of this function, since indeed the SAML modules handle it on their own.